src/core/asyncUtils.ts

```diff
@@ -20,11 +20,11 @@ function asyncUtils(act: Act, addResolver: (callback: () => void) => void): Asyn
       return callbackResult ?? callbackResult === undefined
     }
 
-    const timeoutSignal = createTimeoutController(timeout)
+    const timeoutSignal = createTimeoutController(timeout as number | boolean, false)
```

Comment:
If `interval` is `false` (like it is implicitly for `waitForNextUpdate`, but can also be set by the user for the other utils) you really want to be passing `true` here.  For my local testing, I've been using

```suggestion
    const timeoutSignal = createTimeoutController(timeout as number | boolean, !interval)
```

This way, either `intervalSignal` or `timeoutSignal` will advance the timers, but not both.

Reply:
👍